r/R/dataset-write.R

```diff
@@ -153,6 +153,16 @@ write_dataset <- function(dataset,
     }
   }
 
+  if (!missing(max_rows_per_file) && max_rows_per_group > max_rows_per_file) {
+    if (!missing(max_rows_per_group)) {
+      warning(paste0(c(
+        "'max_rows_per_group' must be less or equal to 'max_rows_per_file'.",
+        "\n'max_rows_per_group' set to value of 'max_rows_per_file'."
```

Comment:
```suggestion
        "`max_rows_per_group` must be less or equal to `max_rows_per_file`.",
        "\n`max_rows_per_group` set to value of `max_rows_per_file`."
```

Nice informative warning here.  Typically, we use backticks (```) to refer to parameter values in these kinds of messages.

How about though, we adjust the conditions under which the change in line 163 below is triggered to only include when `max_rows_per_file` has been set (i.e. isn't missing) *and* `max_rows_per_group` hasn't been changed from the default (i.e. is "missing")?  

I'm hesitant to make too many assumptions about user intentions, but I think we could safely say here that it's in those circumstances the user most likely wants to set the maximum rows per file and just hasn't paid attention to the `max_rows_per_group` parameter, and so we can just update the value silently with no warning.